### PR TITLE
Issue #193 - multiple improvements to dpkt.Packet.__repr__()

### PR DESCRIPTION
**Pull request by obormot**
@kbandla @brifordwylie @saylenty and others - please comment what you think of this. See #193 for details of what I tried to achieve. Here are some examples of packet representation before and after the suggested change (new lines and indent added manually for better looks).

ETH/TCP/IP stack
```python
Ethernet(src='\x00\xa0\xcc;\xbf\xfa', dst='\x00\x00\xc0\x9f\xa0\x97',
         data=IP(src='\xc0\xa8\x00\x02', off=16384, dst='\xc0\xa8\x00\x01', tos=16, sum=29468, len=60, p=6, id=17980,
                 data=TCP(seq=2579865836, off_x2=160, win=32120, sum=57507, dport=23, sport=1550)))

Ethernet(dst='\x00\x00\xc0\x9f\xa0\x97', src='\x00\xa0\xcc;\xbf\xfa',
         data=IP(tos=16, len=60, id=17980, off=16384, p=6, sum=29468, src='\xc0\xa8\x00\x02', dst='\xc0\xa8\x00\x01', opts='',
                 data=TCP(sport=1550, dport=23, seq=2579865836, off=10, win=32120, sum=57507, opts="\x02\x04\x05\xb4\x04\x02\x08\n\x00\x9c'$\x00\x00\x00\x00\x01\x03\x03\x00")))
```

ETH w/ MPLS labels

```python
Ethernet(src='\xcc\x04\x04\xdc\x00\x10', dst='\xcc\x03\x04\xdc\x00\x10', data='...')

Ethernet(dst='\xcc\x03\x04\xdc\x00\x10', src='\xcc\x04\x04\xdc\x00\x10', labels=[(19, 0, 254), (16, 0, 255)], data='...')
```

GRE

```python
GRE(p=34827, flags=12289, data=PPP(p=255, data='...'))))

GRE(flags=12289, p=34827, callid=6144, len=3584, seq=0, data=PPP(p=255, data='...'))))
```

VRRP

```python
VRRP(count=1, advtime=1, sum=47698, vrid=1, priority=100)

VRRP(vrid=1, priority=100, count=1, advtime=1, sum=47698, addrs=['\xc0\xa8\x00\x01'], auth='\x00\x00\x00\x00\x00\x00\x00\x00')
```

STP

```python
STP(max_age=5120, root_id="\x80\x00\x08\x00'\xad\xa3A", bridge_id="\x80\x00\x08\x00'\xad\xa3A", port_id=32769, hello=512, flags=62, v=2, fd=3840, type=2,
    data='...')

STP(v=2, type=2, flags=62, root_id="\x80\x00\x08\x00'\xad\xa3A", bridge_id="\x80\x00\x08\x00'\xad\xa3A", port_id=32769, age=0, hello=2, fd=15,
    data='...')
```

IPv6

```python
IP6(src=' H\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xde\xca', nxt=43, dst=' G\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xca\xfe', hlim=64, plen=60,
    data=TCP(seq=0, win=8192, sum=37247, dport=80, sport=20))

IP6(plen=60, nxt=43, hlim=64, src=' H\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xde\xca', dst=' G\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xca\xfe', p=6,
    extension_hdrs={0: None, 43: IP6RoutingHeader(nxt=6, len=4, segs_left=2, length=40, addresses=[' \x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xde\xca', ' "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xde\xca'],
    data='...'), 44: None, 50: None, 51: None, 60: None},
    data=TCP(sport=20, dport=80, seq=0, win=8192, sum=37247, opts=''))
```

### CODE REVIEW

File: dpkt/dpkt.py [hunk 1]
```diff
@@ -103,9 +106,30 @@ def __getitem__(self, k):
             raise KeyError
 
     def __repr__(self):
```
Comment (brifordwylie) [inline dpkt/dpkt.py]: You are handling more corner cases then I'm aware of here, but this code looks maybe +1 more complicated than it need to be :). Chains https://github.com/SuperCowPowers/chains just does this.

```python
def _make_dict(obj):
    """This method creates a dictionary out of an object"""
    return {key: getattr(obj, key) for key in dir(obj) if not key.startswith('__') and not callable(getattr(obj, key))}
```

Here's the output on TCP
```
Transport: TCP {'off': 8, 'seq': 1669543385, 'off_x2': 128, 'ack': 4092461254, 'win': 4096, 'sum': 37549, 'flags': [], 
'dport': 443, 'urp': 0, 'sport': 53798, 'opts': '\x01\x01\x08\n\x16\x92\x12\xd2to\xf0l', 
'data': '\x17\x03\x03\x00\x1e\x00\x00\x00\x00\x00\x00\x00L3[\xe7\xd4\x0f\xd2\xad\xebU\xe2kA)/\x8e\xeeAk\xba\xa84\x9e', }
```
Obviously this code won't work in this case (you only want keys in \__hdr\__ and order matters...) but maybe some variant of this or something... 

```python
def _make_obj_list(obj):
    """This method creates an ordered list out of an object"""
    return [key: getattr(obj, key) for key ingetattr(self, '__hdr__', []) if not key.startswith('__') and not callable(getattr(obj, key))]
```

**Comment by brifordwylie** [comment]
Perhaps I'm thinking/looking at this wrong but how come I don't see TTL in either output for IPv4? and also Traffic class and Flow Label for IPv6?  Yes I understand these have nothing to do with your changes.. but now that we're looking at it....

**Comment by obormot** [comment]
@brifordwylie the reason you don't see TTLs is most likely 'cause dpkt doesn't display fields when their value == to default value. This change keeps this behavior intact. Try with a different ttl..

**Comment by brifordwylie** [comment]
Okay, that all makes sense... :) I'll let you merge it.

**Pull request merged**